### PR TITLE
Governance

### PR DESCRIPTION
**Pull request by tacaswell**
This is the governance and named roles text from #7 pulled out into its own PR.

### CODE REVIEW

**Comment by sandrotosi** [comment]
first comment! a typo! lol PR title says `Goverance` i guess you meant `Governance`

File: governance.md [hunk 1]
```diff
@@ -197,56 +218,219 @@ interests include, but are not limited to:
...
+the CM is responsible for maintaining:
+
+- the [Ann] mailing list
+- The Matplotlib blog
+- Maintaining a "matplotlib update" slide deck
+- Recruiting people to give talks on Matplotlib at conferences / meetups
+- Matplotlib's social media presence
+
+This communication is primarily broadcast out and intended to engage with
+end-users.
```
Comment (story645) [inline governance.md:395]: change end -users to broader community. Like a good chunk of our social media engagement is core devs on other projects.

File: governance.md [hunk 2]
```diff
@@ -197,56 +218,219 @@ interests include, but are not limited to:
...
+
+**Developer Relations**
+
+In addition to communication to end-users we need to maintain
+communication channels to developers of down-stream libraries and
+other expert users.
```
Comment (story645) [inline governance.md:401]: the discourse is not solely for expert users and like we specifically want folks at all levels 'cause the discourse is also for the folks who won't go near SO. 

The difference between the two community roles as I see it are:
* communications-have a conversation with folks [general] about what's going on w/ the project & what folks can do w/ it & the like - this is more marketing/outreach
* developer relations - have conversations with specific folk about what they're trying to do w/ mpl & want to do - this is more a problem solving/helping folks get acclimated role

File: governance.md [hunk 3]
```diff
@@ -197,56 +218,219 @@ interests include, but are not limited to:
...
+
+- interaction with 3rd party extensions / publicity
```
Comment (story645) [inline governance.md:403]: I think this belongs in comms 'cause like that's the type of stuff the person wearing that hat is good at while like this role doesn't need it. For example I can think of multiple devs who have shown that they're fantastic at all the things on this list except publicity. I think 3rd party extensions go under both 'cause again it's all community.

File: governance.md [hunk 4]
```diff
@@ -9,18 +9,16 @@ section below, is contained in The Project Governance Repository at:
...
-[Matplotlib GitHub organization](https://github.com/matplotlib). Examples
-of Project Software include the matplotlib library for data
-visualization and its associated extensions and dependencies.  The
-Services run by the Project consist of public websites and
-web-services that are hosted under the matplotlib.org domain.
+The Matplotlib Project (The Project) is an open source software project
+affiliated with the [NumFOCUS](https://numfocus.org) Foundation. The goal of The
+Project is to develop open source software and deploy open and public websites
+and services for data visualization. The Software developed by The Project is
+released under the a BSD (or similar permissive) open source license, developed
```
Comment (timhoffm) [inline governance.md:16]:
```suggestion
released under a BSD (or similar permissive) open source license, developed
```

File: governance.md [hunk 5]
```diff
@@ -197,56 +218,219 @@ interests include, but are not limited to:
...
+- notifying down-stream packagers of the release
+- announcing the release (in coordination with the Community Manager)
+
+An individual may be the RM as more than one release series at the
+same time.
+
+
+### API consistency Leader
+
+Matplotlib is constantly making small changes to our API: enhancements
```
Comment (timhoffm) [inline governance.md:320]:
```suggestion
Matplotlib is constantly making small changes to its API: enhancements
```

File: governance.md [hunk 6]
```diff
@@ -197,56 +218,219 @@ interests include, but are not limited to:
...
+- complete
 
+In addition to the docstrings the RDL is responsible for the sphinx build
+machinery and our sphinx extensions.
+
+### Narrative Documentation Leader
+
+In addition to the refernce documentation, Matplotlib has narrative documentation.
+This documentation can take the form of short "cookbook" examples that show an
+end-to-end
```
Comment (timhoffm) [inline governance.md:357]: sentence unfinished

**Comment by tacaswell** [comment]
Maybe the text in the Community Manager description is too detailed / fine-grained? 

On a bit more reflection I realized that this wording bakes an inconsistency into the document (change the DPL by majority vote of SC, governance by 2/3 majority note of SC) so moving them out to their own file.

File: governance.md [hunk 7]
```diff
@@ -197,56 +218,219 @@ interests include, but are not limited to:
...
+- correctly formatted and render as intended
+- technically correct
+- complete
 
+In addition to the docstrings the RDL is responsible for the sphinx build
+machinery and our sphinx extensions.
+
+### Narrative Documentation Leader
+
+In addition to the refernce documentation, Matplotlib has narrative documentation.
```
Comment (jklymak) [inline governance.md:355]:
```suggestion
In addition to the reference documentation, Matplotlib has narrative documentation.
```

File: governance.md [hunk 8]
```diff
@@ -197,56 +218,219 @@ interests include, but are not limited to:
...
+This includes our examples and tutorials that live inside the
+main source repository and longer tutorials that live in other repositories
+in the Matplotlib organization.
+
+The Narrative Documentation Leader is responsible for shepherding all of this
+content including the scope, level, tone, and voice.
+
+### Secretary
+
+- Responsible for ensure that there is an agenda for the weekly meeting at it is
```
Comment (jklymak) [inline governance.md:369]:
```suggestion
- Responsible for ensuring that there is an agenda for the weekly meeting at it is
```

File: governance.md [hunk 9]
```diff
@@ -197,56 +218,219 @@ interests include, but are not limited to:
...
+  followed.
+- Responsible for maintaining the weekly notes.
+- Responsible for ensuring that decisions taken by vote in the SC are properly
```
Comment (jklymak) [inline governance.md:372]: This is a confusing responsibility, unless the Secretary will be part of the SC.

Comment (tacaswell) [reply governance.md:372]: Good point, moved this to PL.

**Comment by story645** [comment]
> Maybe the text in the Community Manager description is too detailed / fine-grained?
Yes/No in that it's got more of an umbrella role than the others, but I think it's fine to pull the description back to something like 
* does evangelism, outreach, and support. 
*responsible for setting the tone and maintaining the venues for conversation w/ the community

**Comment by jklymak** [comment]
I'm having some trouble placing this in the framework of other organizations I'm familiar with.  The steering council seems like an executive, but no roles are discussed in the context of the steering council.  So there is no Treasurer, Secretary, or any other responsibility designated to the steering council.   Similarly, it would be unusual to define the president, and then the group that he is president of - the president is typically just the leader of the executive.  

I think the current setup is a bit unclear who has ultimate authority.  I think it should be the steering council, and that the PL's authority should devolve from the steering council.  I appreciate that is a change, but I think it would be more robust to problems with future PLs.  I know you claim forking is the way to change the PL, but if a future PL takes the project down some crazy route, its not particularly acceptable for them to take the `matplotlib` name with them, and it doesn't seem like it should be allowed to happen if there is a reasonable Steering Council in place.

**Comment by tacaswell** [comment]
That is a fair point, however I would prefer to keep this discussion about just the Deputy Lead roles (and renaming BDFL -> PL) rather than completely re-do the governance structure.  I can remove all of the changes to the SC if that would make it clearer.

Historically we have been a BDFL / single lead + rough consensus style project which in some sense is, as long as you trust the PL (and I trust me and hope y'all also trust me), the simplest and clearest ultimate authority ;).

I think one of the worst failure modes here are a complicated governance document that we don't actually follow.  For better-or-worse, we have an existing (and seemingly functioning) process that I worry if we try to make too big of a change we will revert back to (and evidence of this being a problem I cite my failure to fully stand up the SC yet).  There have been a couple of projects (notably skimage and cpython) that have moved from single leader -> committee that we can use as a model.  My preference here is that we actually stand up a full-sized (bigger than 3) SC and then writing them selves into full existence / authority.

**Comment by jklymak** [comment]
> ... and then writing them selves into full existence / authority.

Sure, if they have that authority.  

The problem is not whether we trust you, its whether you get hit by a bus tomorrow, then what happens?

**Comment by tacaswell** [comment]
> Sure, if they have that authority.

They do because the document has a method of changing its self (via 2/3 vote of current SC)

**Comment by jklymak** [comment]
> They do because the document has a method of changing its self (via 2/3 vote of current SC)

... which the PL can veto?  

> The full Steering Council must vote and at least 2/3 of the votes must be positive to carry out the proposed action (fractions of a vote rounded up to the nearest integer). Since the PL holds ultimate authority in The Project, the PL has authority to act alone in accepting or rejecting changes or overriding Steering Council decisions.

File: deputy_project_leads.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+- ensuring the whats new, API changes, and release notes are up to date
+- the timing of the releases
+- what changes should or should not be backported from the master
+  branch
+- rebuilding and publishing the website
+- publishing sdist and wheels to pypi
+- notifying down-stream packagers of the release
+- announcing the release (in coordination with the Community Manager)
+
+An individual may be the RM as more than one release series at the
```
Comment (dopplershift) [inline deputy_project_leads.md:28]: "...may be the RM for more than..."?

File: deputy_project_leads.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+- Responsible for ensuring that there is an agenda for the weekly
+  meeting at it is followed.
+- Responsible for maintaining the weekly notes.
+
+
+## Community Manager
+
+The true strength of Matplotlib and why it has had such longevity as a project
+is the community of people around the code.  That community needs to be
+maintained.  The Community Manager (CM) is the is a catch-all for several very
```
Comment (dopplershift) [inline deputy_project_leads.md:89]: "is the is a catch-all" -> "is catch-all position"?

File: governance.md [hunk 12]
```diff
@@ -30,7 +28,7 @@ Development Team (MDT)" in the project license.  Anyone can be a
 Contributor. Contributors can be affiliated with any legal entity or
 none. Contributors participate in the project by submitting, reviewing
 and discussing GitHub Pull Requests and Issues and participating in
-open and public Project discussions on GitHub, Google+, Hackpad,
+open and public Project discussions on GitHub, discourse, Hackpad,
```
Comment (dopplershift) [inline governance.md:31]: why not "Discourse"?

Comment (story645) [reply governance.md:31]: also hackmd not hackpad

**Comment by tacaswell** [comment]
@efiring @dopplershift Can you please approve and merge this?  Once it hits master it will be "in force".

File: deputy_project_leads.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+The true strength of Matplotlib and why it has had such longevity as a
+project is the community of people around the code.  That community
+needs to be maintained.  The Community Manager (CM) is a catch-all
+position for several very diverse tasks and this role may be split in
+the future and may want to enlist further assistants.
+
+**Communications**
+
+We need to maintain communication channels with the wider world.  To
+this end the CM is responsible for maintaining.  This communication is
```
Comment (dopplershift) [inline deputy_project_leads.md:87]: "To this end the CM is responsible for maintaining." maintaining what?

File: deputy_project_leads.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+primarily broadcast out and intended to engage with the wider
+community.  The
+
+- the [Ann] mailing list
```
Comment (dopplershift) [inline deputy_project_leads.md:91]: This is the only one not capitalized.

File: deputy_project_leads.md [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+- Matplotlib's social media presence
+
+**Developer Relations**
+
+In addition to communication to end-users we need to maintain
+communication channels to developers of down-stream libraries and
+other expert users.
+
+- interaction with 3rd party extensions / publicity
+- recruiting new contributors
```
Comment (dopplershift) [inline deputy_project_leads.md:104]: Again, inconsistent capitalization.

File: people.md [hunk 16]
```diff
@@ -1,8 +1,19 @@
 # Steering Council and Institutional Partners
 
-## Benevolent Dictator for Life
+## Project Lead
 
-Thomas Caswell is the Benevolent Dictator for Life (BDFL).
+Thomas Caswell is the currenwt Project Lead
```
Comment (dopplershift) [inline people.md:5]: "currenwt" -> "current"

**Comment by tacaswell** [comment]
Fixed the spelling, we might want to tighten up the whole CM section to be closer in length to the others.

File: governance.md [hunk 17]
```diff
@@ -83,111 +81,135 @@ Community.
...
+
+The PL can appoint their successor, but it is expected that the Steering Council
+would be consulted on this decision. If the PL is unable to appoint a successor,
+the Steering Council will make a suggestion or suggestions to the Main NumFOCUS
+Board. While the Steering Council and Main NumFOCUS Board will work together
+closely on the PL selection process, the Main NumFOCUS Board will make the final
+decision.  The NumFOCUS board may in extenuating circumstances remove the PL
+from the project.
+
+To ensure the benevolence of the PL, The Project encourages others to fork the
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply governance.md:116]: It was in the IPython documentation, but I take your (and @jklymak 's) point that at this point any given person is far more replaceable than then project name / brand so this is not actually a plausible recourse anymore.

[the remark this answers is not in the document]
Comment (efiring) [reply governance.md:116]: And it makes no sense for the governance document to either encourage, or discourage forking; everyone knows it's an option, and they can do it if they choose without needing a line in the governance doc to encourage it.

File: governance.md [hunk 18]
```diff
@@ -83,111 +81,135 @@ Community.
...
+additional member must be recruited.
+
+All former Council members can be considered for membership again at any time in
+the future, like any other Project Contributor. Retired Council members will be
+listed on the project website, acknowledging the period during which they were
+active in the Council.
+
+The Council reserves the right to eject current Members, other than the PL, if
+they are deemed to be actively harmful to the project’s well-being, and attempts
+at communication and conflict resolution have failed, from The Project.
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply governance.md:208]: I think this was meant to eject people from the SC but not from the project overall, but I am not sure the middle option (stay in project but ejected from SC?), changing to be fully eject from the project.

[the remark this answers is not in the document]
Comment (efiring) [reply governance.md:208]: Is it even defined for a person to be in, or out of, The Project?  I don't think we define project membership as such; there are Contributors, Developers, etc. but not Members, I think.

[the remark this answers is not in the document]
Comment (tacaswell) [reply governance.md]: > The Council reserves the right to eject anyone, including the Council
Members, from the project's online spaces if they are deemed to be
actively harmful to the project’s well-being, and attempts at
communication and conflict resolution have failed, by a super majority vote.

We can ban people from github / discourse / mailing lists / gitter if they are being disruptive.

File: governance.md [hunk 19]
```diff
@@ -83,111 +81,135 @@ Community.
...
 
 ### Conflict of interest
 
-It is expected that the BDFL and Council Members will be employed at a wide
+It is expected that the PL, DPLs, and Council Members will be employed at a wide
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply governance.md:214]: I think I got all of these?

**Comment by tacaswell** [comment]
Well, I was going to do follow ups, but it's merged so :tada:

File: deputy_project_leads.md [hunk 20]
```diff
@@ -79,36 +79,6 @@ The true strength of Matplotlib and why it has had such longevity as a
...
-- Managing the discourse
-- [user]/[dev] mailing list moderation
-
-
-**Event and Meeting planning**
-
-From time to time Matplotlib as an organization has in-person
-meetings.
+the future and may want to enlist further assistants.  The CM is responsible
+for Matplotlib's evangelism, outreach, and user support as well as mainitaining
```
Comment (story645) [inline deputy_project_leads.md:83]: for community outreach and support, as well as for publicizing what's new and useful in the project. 

(I hate the word evangelism)

Comment (efiring) [reply deputy_project_leads.md:83]: @story645 Your review came after I merged the PR, but please go ahead and put in a PR with this fix and any other fine-tuning you see.

**Comment by efiring** [comment]
Sorry if I jumped the gun, but I'll blame @dopplershift--I inferred from his review that it was time to just get this in.

**Comment by dopplershift** [comment]
I'll take the blame. I didn't want to grammar this to death, especially for things that are not really governing (like verbiage describing DPL roles), since we've spent so long getting things going governance-wise.

**Comment by story645** [comment]
I just didn't see the merge, it's all fine.

**Pull request merged**